src/core/memory.d

```diff
@@ -1012,13 +1012,19 @@ void __delete(T)(ref T x) @system
 
     static if (is(T == interface) || is(T == class))
     {
-        GC.free(cast(void*) x);
-        x = null;
+        if (x !is null)
+        {
+            GC.free(cast(void*) x);
+            x = null;
+        }
     }
     else static if (is(T == U2*, U2) || is(T : E2[], E2))
     {
-        GC.free(&x);
-        x = null;
+        if (x !is null)
+        {
+            GC.free(&x);
```

Comment:
This seems wrong, shouldn't it be `GC.free(x)` ?

Reply:
Yeah, looks like I was a bit too fast. Thanks!